runtimes/runtime-blocking-bootstrap/src/main/java/com/flipkart/aesop/runtime/bootstrap/consumer/DefaultBlockingEventConsumer.java

```diff
@@ -47,11 +47,11 @@ public DefaultBlockingEventConsumer(int numberOfPartition, int executorQueueSize
 	{
 		this.eventConsumer = eventConsumer;
 		this.numberOfPartition = Math.min(numberOfPartition, Runtime.getRuntime().availableProcessors());
-		BlockingQueue<Runnable> queue = new ArrayBlockingQueue<Runnable>(executorQueueSize);
 
 		LOGGER.info("numberOfPartition used: " + numberOfPartition);
 		for (int i = 0; i < numberOfPartition; i++)
 		{
+			BlockingQueue<Runnable> queue = new ArrayBlockingQueue<Runnable>(executorQueueSize);
```

Comment:
Context - The idea of having multiple single-threaded executors was so that we should be able to send insert/updates for a entity to same executor (choosing executor via hash of primary key % number of executors), so that the order of updates remains preserved.

The error here is that same instance of blocking queue is shared by all executors. Now, even though executor is selected via "hash mod" logic, all executors will block on same queue, removing the guarantee of same entity being processed by same executor always.

The fix is simple. Pass a new instance of blocking queue to every executor.